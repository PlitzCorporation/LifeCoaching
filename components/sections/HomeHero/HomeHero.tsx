import Image from 'next/image';

import altCreator from '@/helpers/altCreator';
import { WhiteThickLine } from '@/components/ui/decorations';

export interface HomeHeroProps {
	title: string;
	subtitle: string;
	content: string;
	image: string;
}

const HomeHero = (props: HomeHeroProps) => {
	const { title, subtitle, content, image } = props;

	const altText = altCreator(image);

	return (
		<div className="relative h-max md:h-200 lg:h-250 overflow-hidden 3xl:max-w-[95vw] mx-auto bg-plitz-primary">
			<div className="relative lg:absolute z-10 top-0 bottom-0 w-full">
				<div className="plitz-container flex items-center h-full">
					<div className="grid grid-cols-1 lg:grid-cols-2 gap-4">
						<div className="min-h-9"></div>
						<div className="flex flex-col text-left drop-shadow-plitz-darker py-10">
							<WhiteThickLine />
							<h1 className="text-white lg:drop-shadow-plitz-titles mt-6 mb-4">
								{title}
							</h1>
							<h2 className="leading-tight text-white lg:drop-shadow-plitz-titles text-balance mb-8">
								{subtitle}
							</h2>
							<div
								dangerouslySetInnerHTML={{ __html: content }}
								className="text-white lg:drop-shadow-plitz-titles text-balance *:text-2xl lg:*:text-4xl"
							/>
						</div>
					</div>
				</div>
			</div>

			<div className="relative h-96 lg:absolute lg:h-full w-full z-0">
				<Image
					src={`${process.env.NEXT_PUBLIC_CMS}/${image}`}
					alt={altText}
					width={0}
					height={0}
					sizes="100vw"
					priority
					className="object-cover object-bottom min-w-full min-h-full max-w-full max-h-full"
				/>
			</div>
		</div>
	);
};

export default HomeHero;
