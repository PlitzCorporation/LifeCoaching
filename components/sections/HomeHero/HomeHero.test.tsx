import { render } from '@testing-library/react';
import HomeHero, { HomeHeroProps } from './HomeHero';

jest.mock('next/image', () => ({
	__esModule: true,
	// eslint-disable-next-line @typescript-eslint/no-explicit-any
	default: (props: any) => {
		// eslint-disable-next-line @next/next/no-img-element, jsx-a11y/alt-text
		return <img {...props} priority="true" />;
	},
}));

describe('HomeHero', () => {
	const OLD_ENV = process.env;

	beforeAll(() => {
		jest.resetModules();
		process.env.NEXT_PUBLIC_CMS = 'https://thedavid.plitz7.com';
	});

	afterAll(() => {
		process.env = OLD_ENV;
	});

	const props: HomeHeroProps = {
		title: 'Hero Title',
		subtitle: 'Hero Subtitle',
		content: 'Hero Content',
		image: 'uploads/Reason4Hope/image.jpg',
	};

	it('renders the correct hero content', () => {
		const { container } = render(<HomeHero {...props} />);
		expect(container.firstChild).toMatchSnapshot();
	});
});
