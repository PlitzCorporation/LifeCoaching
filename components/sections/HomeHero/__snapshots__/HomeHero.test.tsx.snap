// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeHero renders the correct hero content 1`] = `
<div
  class="relative h-max md:h-200 lg:h-250 overflow-hidden 3xl:max-w-[95vw] mx-auto bg-plitz-primary"
>
  <div
    class="relative lg:absolute z-10 top-0 bottom-0 w-full"
  >
    <div
      class="plitz-container flex items-center h-full"
    >
      <div
        class="grid grid-cols-1 lg:grid-cols-2 gap-4"
      >
        <div
          class="min-h-9"
        />
        <div
          class="flex flex-col text-left drop-shadow-plitz-darker py-10"
        >
          <div
            class="block w-32 h-1 bg-gradient-to-r from-white/5 to-white/100"
          />
          <h1
            class="text-white lg:drop-shadow-plitz-titles mt-6 mb-4"
          >
            Hero Title
          </h1>
          <h2
            class="leading-tight text-white lg:drop-shadow-plitz-titles text-balance mb-8"
          >
            Hero Subtitle
          </h2>
          <div
            class="text-white lg:drop-shadow-plitz-titles text-balance *:text-2xl lg:*:text-4xl"
          >
            Hero Content
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="relative h-96 lg:absolute lg:h-full w-full z-0"
  >
    <img
      alt="uploads/Reason4Hope/image"
      class="object-cover object-bottom min-w-full min-h-full max-w-full max-h-full"
      height="0"
      priority="true"
      sizes="100vw"
      src="https://thedavid.plitz7.com/uploads/Reason4Hope/image.jpg"
      width="0"
    />
  </div>
</div>
`;
